integration/network_test.go

```diff
@@ -0,0 +1,861 @@
+			spName:       "network1",
+			pingHostName: "DNS1",
+			expectedRunContainerStatus: func(err error) bool {
+				return err != nil
+			},
+		},
+		{
+			testName:     "ping using instance path between different subnet",
+			spName:       "network1",
+			pingHostName: "0.subject1.serverDNS1.network0",
+			expectedRunContainerStatus: func(err error) bool {
+				return err != nil
+			},
+		},
+	}
+
+	for i, tCase := range cases {
+		t.Run(tCase.testName, func(t *testing.T) {
+			instanceID := fmt.Sprintf("instance%d", i)
+			containerPath := path.Join(tmpDir, instanceID)
+
+			if err = createOCIContainer(
+				manager, containerPath, instanceID,
+				[]string{"ping", tCase.pingHostName, "-c1", "-w5"}, true); err != nil {
+				t.Errorf("Can't create container: %v", err)
+			}
+
+			if err = manager.AddInstanceToNetwork(instanceID, tCase.spName, networkmanager.NetworkParams{
+				ResolvConfFilePath: path.Join(containerPath, "etc", "resolv.conf"),
+			}); err != nil {
+				t.Errorf("Can't add instance to network: %v", err)
+			}
+
+			if err = runOCIContainer(containerPath, instanceID); !tCase.expectedRunContainerStatus(err) {
+				t.Error("Unexpected result run container")
+			}
+		})
+	}
+
+	for i, tCase := range cases {
+		if err = manager.RemoveInstanceFromNetwork(fmt.Sprintf("instance%d", i), tCase.spName); err != nil {
+			t.Errorf("Can't remove instance from network: %v", err)
+		}
+	}
+
+	for _, tCase := range casesInfinityRunningInstance {
+		if err = manager.RemoveInstanceFromNetwork(tCase.instanceName, tCase.spName); err != nil {
+			t.Errorf("Can't remove instance from network: %v", err)
+		}
+
+		if err = killOCIContainer(tCase.instanceName); err != nil {
+			t.Errorf("Can't kill container: %v", err)
+		}
+	}
+}
+
+func TestBandwidth(t *testing.T) {
+	manager, err := networkmanager.New(&config.Config{WorkingDir: tmpDir}, nil)
+	if err != nil {
+		t.Fatalf("Can't create network manager: %v", err)
+	}
+	defer manager.Close()
+
+	container0Path := path.Join(tmpDir, "instance0")
+
+	var (
+		setULSpeed uint64 = 1000
+		setDLSpeed uint64 = 4000
+	)
+
+	if err := createOCIContainer(manager, container0Path, "instance0", []string{"iperf", "-s"}, false); err != nil {
+		t.Fatalf("Can't create container: %v", err)
+	}
+
+	if err := manager.AddInstanceToNetwork("instance0", "network0", networkmanager.NetworkParams{
+		IngressKbit: setDLSpeed,
+		EgressKbit:  setULSpeed,
+	}); err != nil {
+		t.Fatalf("Can't add instance to network: %v", err)
+	}
+
+	go func() {
+		_ = runOCIContainer(container0Path, "instance0")
+	}()
+
+	time.Sleep(1 * time.Second)
+
+	ip, err := manager.GetInstanceIP("instance0", "network0")
+	if err != nil {
+		t.Errorf("Can't get ip address from instance: %v", err)
+	}
+
+	output, err := exec.Command("iperf", "-c", ip, "-d", "-r", "-t5", "-yc").CombinedOutput()
+	if err != nil {
+		t.Errorf("iperf failed: %v", err)
+	}
+
+	ulSpeed := -1.0
+	dlSpeed := -1.0
+
+	lines := strings.Split(string(output), "\n")
```

Comment:
Move is inplace. No need for intermediate var.

Reply:
Done